query-engine/connector-test-kit/src/test/scala/writes/relations/RelationDefaultsSpec.scala

```diff
@@ -0,0 +1,125 @@
+  "Not providing a value for a required relation with multiple fields with one default value" should "not work" taggedAs IgnoreSQLite in {
+    val schema =
+      """model List{
+        |   id        Int  @id @default(autoincrement())
+        |   uList     String? @unique
+        |   todoId    Int @default(1)
+        |   todoName  String
+        |   todo      Todo   @relation(fields: [todoId, todoName], references: [id, uTodo])
+        |}
+        |
+        |model Todo{
+        |   id     Int @default(autoincrement())
+        |   uTodo  String
+        |   lists  List[]
+        |
+        |   @@id([id, uTodo])
+        |}"""
+
+    val project = SchemaDsl.fromStringV11() { schema }
+
+    database.setup(project)
+
+    // Setup
+    server.query(s"""mutation {createList(data: {uList: "A", todo : { create: {uTodo: "B"}}}){id}}""", project)
+
+    val result = server.query(s"""query{lists {uList, todo {uTodo}}}""", project)
+    result.toString should equal("""{"data":{"lists":[{"uList":"A","todo":{"uTodo":"B"}}]}}""")
+
+    server.query(s"""query{todoes {uTodo}}""", project).toString should be("""{"data":{"todoes":[{"uTodo":"B"}]}}""")
+
+    countItems(project, "lists") should be(1)
+    countItems(project, "todoes") should be(1)
+
+    // Check that we still need to provide the name
+    server.queryThatMustFail(
+      s"""mutation { createList(data: {uList: "listWithTodoOne"}) { id todo { id } } }""",
+      project,
+      errorCode = 2012,
+      errorContains = "Missing a required value at `Mutation.createList.data.ListCreateInput.todo`",
+    )
+
+    countItems(project, "lists") should be(1)
+  }
+
+  // We ignore SQLite because a multi-column primary key cannot have an autoincrement column on SQLite.
+  "Not providing a value for one field with a default in a required relation with multiple fields" should "work" taggedAs IgnoreSQLite in {
+    val schema =
+      """model List{
+        |   id        Int  @id @default(autoincrement())
+        |   uList     String? @unique
+        |   todoId    Int @default(1)
+        |   todoName  String
+        |   todo      Todo   @relation(fields: [todoId, todoName], references: [id, uTodo])
+        |}
+        |
+        |model Todo{
+        |   id     Int @default(autoincrement())
+        |   uTodo  String
+        |   lists  List[]
+        |
+        |   @@id([id, uTodo])
+        |}"""
+
+    val project = SchemaDsl.fromStringV11() { schema }
+
+    database.setup(project)
+
+    // Test that we can still create with the value without default only
+    val result2 =
+      server.query(s"""mutation { createList(data: {uList: "listWithTodoOne", todo: { create: { uTodo: "abcd" } }}) { id todo { id } } }""", project)
+    result2.toString should equal("""{"data":{"createList":{"id":1,"todo":{"id":1}}}}""")
+
+    countItems(project, "lists") should be(1)
+    countItems(project, "todoes") should be(1)
+  }
+
```

Comment:
I'm thinking we could have one more test case: multi-column PK, with all the columns having a default. Then the input should be optional as well. Does it make sense to write it?

Reply:
yes

Reply:
Imo the convention in this test file is that fields prefixed with `u` like `uList` and `uTodo` are unique. I just had problems understanding the test because `uTodo` is not unique. Don't know if that warrants a change but just wanted to mention it.

Reply:
I started renaming these to just `name`, I'll do that across the whole file, thanks for the comment :+1: